ethcore/src/client/mod.rs

```diff
@@ -65,20 +65,48 @@ pub trait BlockChainClient : Sync + Send {
 	/// Get block total difficulty.
 	fn block_total_difficulty(&self, id: BlockID) -> Option<U256>;
 
-	/// Get address nonce.
-	fn nonce(&self, address: &Address) -> U256;
+	/// Attempt to get address nonce at given block.
+	/// May not fail on BlockID::Latest.
+	fn nonce(&self, address: &Address, id: BlockID) -> Option<U256>;
+
+	/// Get address nonce at the latest block's state.
+	fn nonce_latest(&self, address: &Address) -> U256 {
```

Comment:
I think the most common case is to use `nonce_latest` which should stay `nonce` imho.

Can we have `nonce_at_block` or sth like this instead?